Agrume/AgrumeCell.swift

```diff
@@ -482,4 +495,27 @@ extension AgrumeCell: UIScrollViewDelegate {
       dismiss()
     }
   }
+  
+  @available(iOS 16, *)
+  private func analyzeImage(_ image: UIImage) async {
+    guard ImageAnalyzer.isSupported else {
+      return
+    }
+
+    let analyzer = ImageAnalyzer()
+    let interaction = await MainActor.run {
+      let interaction = ImageAnalysisInteraction()
+      imageView.addInteraction(interaction)
+      return interaction
+    }
+    let configuration = ImageAnalyzer.Configuration([.text, .machineReadableCode])
+    do {
+      let analysis = try await analyzer.analyze(image, configuration: configuration)
+      await MainActor.run {
+        interaction.analysis = analysis
+        interaction.preferredInteractionTypes = .automatic
+      }
+    } catch {
+    }
+  }
```

Comment:
What if we changed the method to not be async and instead encapsulate that inside of the method itself. Then we don't need the `await MainActor.run` dance to construct the interaction. We could then also make use of `Task { @MainActor in`

Another alternative would be to keep the method `async` but annotate it to run on the `MainActor`:

```swift
@MainActor
private func analyzeImage(_ image: UIImage) async {
  …
}
```

```suggestion
  private func analyzeImage(_ image: UIImage) {
    guard ImageAnalyzer.isSupported else {
      return
    }
    let interaction = ImageAnalysisInteraction()
    imageView.addInteraction(interaction)

    let analyzer = ImageAnalyzer()
    let configuration = ImageAnalyzer.Configuration([.text, .machineReadableCode])

    Task { @MainActor in
      do {
        let analysis = try await analyzer.analyze(image, configuration: configuration)
        interaction.analysis = analysis
        interaction.preferredInteractionTypes = .automatic
      } catch {
        print(error.localizedDescription)
      }
    }
  }
```